src/sp/it/pl/main/App.kt

```diff
@@ -241,11 +242,12 @@ class App: Application(), Configurable<Any> {
     @F val plugins = PluginManager()
 
     init {
-        AppUtil.APP = this.takeIf { AppUtil.APP==null } ?: fail("Multiple application instances disallowed")
+        APP = this.takeUnless { isAPPInitialized } ?: fail("Multiple application instances disallowed")
```

Comment:
I see, this is the reason for including the isAppInitialized field in AppUtils. The intention I think was: "hey lets name the predicate to be more readable, so lets make it a method, hey then lets push itto AppUtils so everyone can use it if necessary". This is good intention, but ignored the fact that nobody else should ask if APP field is null, because nobody should see it as null. Because of that we are exposing unnecessary (and confusing) method/field to the whole world as well as diminish code cohesion - the check APP==null belongs right next to where APP has its value assigned. It was done deliberately so.

Reply:
No, the reason was that you can't access `::APP.isInitialized` outside of the file where `APP` is declared, otherwise I wouldn't have created the method at all ^^

Reply:
Oh my... Im even more convinced now. That file is an abomination. Fortunately it has no purpose now.